src/inmanta/agent/handler.py

```diff
@@ -288,15 +313,37 @@ def action_id(self) -> uuid.UUID:
     def status(self) -> Optional[const.ResourceState]:
         return self._status
 
+    @property
+    def resource_state(self) -> Optional[ContextResourceState]:
+        return self._resource_state
+
     @property
     def logs(self) -> list[data.LogLine]:
         return self._logs
 
     def set_status(self, status: const.ResourceState) -> None:
         """
-        Set the status of the handler operation.
+        Set the status of the handler operation and translate it to ContextResourceState
```

Comment:
the old _status is the `const.ResourceState` while the new `_resource_state` is the `HandlerResourceState`.
`_resource_state` is the selection of states that we expect to see/set on the handler which is a subset of `const.ResourceState` with the inclusion of `skipped_for_dependency`. 
We translate 1:1  `_resource_state` to `status` (making `skipped_for_dependency`, `skipped`)
And we translate `_status` to `resource_state` when we can, otherwise we set `_resource_state` as `failed` because, to my understanding, we don't expect to see the other states in the handler.

Related to this discussion:
https://inmanta.slack.com/archives/CKRF0C8R3/p1732523218360489?thread_ts=1732522716.782239&cid=CKRF0C8R3

Reply:
Ok. Somehow I had completely missed that this method still set the old status field as well, that's what confused me. I think the current implementation is fine, with two small notes:
- I'd prefer to set the new status explicitly to failed, rather than to rely on what happens later
- remember to remove the comment

Reply:
> I'd prefer to set the new status explicitly to failed, rather than to rely on what happens later

I'm not following this comment. What do you mean about explicitly setting the new status to failed and how does it differ from my approach?

Reply:
In the `except` block you now simply log a warning and stop there, leaving the new status at `None`.

Reply:
https://github.com/inmanta/inmanta-core/pull/8383/files#diff-6fe603b0e2ead0d2899810275d0ed57935e40cd1daf59b0424fb3cc0d5c6e53dR323-R333

I now am also setting the status to failed.